.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -20,13 +20,15 @@ npm packages changed:
 - [ ] responsibleai/localization
 - [ ] responsibleai/mlchartlib
 - [ ] responsibleai/model-assessment
+- [ ] No npm packages were changed.

```

Comment:
is there a better type of checkbox we can use?
this is kind of annoying to see on PRs:
![image](https://user-images.githubusercontent.com/24683184/148388306-97d915ce-13f7-4e84-a023-849e220a379f.png)

Reply:
Oh I hadn't seen that before! This is kind of annoying. Let me check! I believe ✔️ `:heavy_check_mark:` should work, but I'll have to try!

Reply:
❌  for "no" and ✔️ for "yes" should be good! Updating this right now! LMK what you think.

Reply:
hmm that's kind of annoying because you can't just check the "viewed" version, you have to modify the underlying text, but maybe that's the best we can do on github

Reply:
huh, maybe what we should do is keep the checkboxes, but rephrase them to be tasks, like:

- [ ] Documentation was updated if it was needed.
- [ ] New tests were added or changes were manually verified.

But then for the "npm packages changed" I guess it doesn't really make sense.  Actually, is there a reason we should keep track of the npm packages that were changed?  The release process doesn't really care about this, does it?
Similarly, I'm not sure about python packages changed - would it help to keep track of that?

Reply:
I don't know, maybe this PR is fine.  Maybe we just need to see how contributors use it and get feedback from them.

Reply:
True, but the "viewed" version is identical with the task list, so you can't have both that and then not the task list in the PR overview. 🙁 

To be honest, the most important part of this PR template for me is to get all of us to write nice descriptions for PRs and include screenshots where useful. I know we haven't been great on this in the past, but it just helps so much and isn't a lot of work for the PR author. [I would add that your descriptions are typically top notch @imatiach-msft ! If all of us did it like you we wouldn't need a template as a reminder at all...]

In addition, we've added lots of features without tests in the past. The corresponding section is there as a reminder to make sure we don't do that anymore. A simple checkbox like yours might accomplish the same. Same for documentation.

The section about packages is there with the hope to eventually be able to auto-extract information on which changes happened to which packages. Perhaps that's not as important, at least not urgently or immediately important... (?)

So perhaps let me minimize it to 
- description
- screenshots
- the two checkboxes you mentioned above.
I'll update the PR later today.

Reply:
I kind of like the checkbox format better. It gives a very good high level view of the scope of changes in the PR. 

I feel the description writing varies with author to author and also with the time pressure. It is just really hard to enforce on every body. With the checkbox system, at least there is some control over the descriptions.

Reply:
Couldn't agree more (in fact that's my motivation for the PR). However, the fact that it displays it as tasks (like in Ilya's screenshot) worries me. Could be interpreted as "there's a lot of work to be done" rather than the actual purpose of the checkboxes.

One alternative I could think of: add the various packages as a simple list (no checkboxes) and the author has to delete all the ones they're not editing so that only the right ones remain. That's easy enough, just like clicking the boxes. 

I do like the simplified single box for tests and documentation. They're really just a nudge in the right direction after all.

I'll definitely wait to hear back from @gaugup 🙂

Reply:
@romanlutz I like "
One alternative I could think of: add the various packages as a simple list (no checkboxes) and the author has to delete all the ones they're not editing so that only the right ones remain. That's easy enough, just like clicking the boxes.". 

I don't however see the corresponding change in this PR.

Reply:
Yes, that was only an idea, but I didn't put it in yet because I'm not entirely sure of the value proposition myself. Initially I had thought perhaps we can simplify release notes by generating them off of the checkboxes, but if people just delete items off of this list and perhaps even add new ones that makes it tricky. Perhaps it's best to keep it simple and without the section describing the areas changed. The description of the PR can take care of that after all. Do you see value in the programmatic generation of release notes?

Reply:
Since there was no more input I'm going to assume it's fine. It doesn't seem like anyone particularly like the additional work the current template caused, so let's see how the new one does and we can always change it.